### PR TITLE
Market indexing helper functions

### PR DESCRIPTION
**Pull request by 0xtristan**

### CODE REVIEW

File: src/market.ts [hunk 1]
```diff
@@ -81,23 +81,56 @@ export class ZetaGroupMarkets {
...
   }
 
+  /**
+   * Returns the options market given an expiry index and options kind.
+   */
+  public getOptionsMarketByExpiryIndex(expiryIndex: number, kind: Kind): Market[] {
+    let markets = this.getMarketsByExpiryIndex(expiryIndex);
+    switch (kind) {
+        case Kind.CALL:
+            return markets.slice(0, this.productsPerExpiry());
```
Comment (xjcaa) [inline src/market.ts:96]: slice.(0, 0+NUM_STRIKES)
and 
slice(NUM_STRIKES, NUM_STRIKES * 2)

Comment (xjcaa) [reply src/market.ts:96]: You want [0..10] and [11..21]

File: src/market.ts [hunk 2]
```diff
@@ -81,23 +81,56 @@ export class ZetaGroupMarkets {
...
     let market = markets[markets.length - 1];
     if (market.kind != Kind.FUTURE) {
       throw Error("Futures market kind error");
     }
     return market;
   }
 
+  public getMarketByExpiryKindStrike(expiryIndex: number, kind: Kind, strike?: number): Market | undefined {
+      let markets = this.getMarketsByExpiryIndex(expiryIndex);
+      let marketsKind = undefined;
```
Comment (xjcaa) [inline src/market.ts:118]: `let marketsKind: Array<Market>`

File: src/market.ts [hunk 3]
```diff
@@ -81,23 +81,56 @@ export class ZetaGroupMarkets {
...
+              throw new Error("Strike must be specified for options markets");
+          }
+          marketsKind = this.getOptionsMarketByExpiryIndex(expiryIndex, kind);
+      } else if (kind === Kind.FUTURE) {
+          return this.getFuturesMarketByExpiryIndex(expiryIndex);
+      } else {
+          throw new Error("Only CALL, PUT, FUTURE kinds are supported");
+      }
+
+      return marketsKind.filter(x => x.strike === strike)[0];
```
Comment (xjcaa) [inline src/market.ts:130]:
```
let markets = marketsKind.filter(...);
return markets.length == 0 ? undefined : markets[0];
```

File: src/market.ts [hunk 4]
```diff
@@ -81,23 +81,56 @@ export class ZetaGroupMarkets {
...
+  }
+
```
Comment (xjcaa) [inline src/market.ts:132]: whitespace

File: src/market.ts (hunk 1 above)
Comment (0xtristan) [reply src/market.ts:96]: Oh yeah my bad I used the wrong thing lol

File: src/market.ts (hunk 3 above)
Comment (0xtristan) [reply src/market.ts:130]: Won't this return undefined anyway?

**Review by xjcaa — APPROVED** [review]

**Pull request merged**